src/bridge/mod.rs

```diff
@@ -253,42 +248,65 @@ impl<IS: AsyncRead + AsyncWrite + 'static> Bridge<IS> {
         }
     }
 
-    fn poll_irc(&mut self) -> Poll<(), io::Error> {
+    pub async fn poll_irc(&mut self) -> Result<(), io::Error> {
         // Don't handle more IRC messages until we have done an initial sync.
         // This is safe as we will get woken up by the sync.
         if self.is_first_sync {
-            return Ok(Async::NotReady);
+            return Ok(());
         }
 
         loop {
-            if let Some(line) = futures::try_ready!(self.irc_conn.poll()) {
-                self.handle_irc_cmd(line);
+            let poll_response = match self.irc_conn.as_mut().poll().await? {
+                Poll::Ready(x) => x,
+                Poll::Pending => return Ok(()),
+            };
+
+            if let Some(line) = poll_response {
+                self.handle_irc_cmd(line).await;
             } else {
                 self.closed = true;
-                self.stop();
-                return Ok(Async::Ready(()));
+                return Ok(());
             }
         }
     }
 
-    fn poll_matrix(&mut self) -> Poll<(), io::Error> {
-        loop {
-            if let Some(sync_response) = futures::try_ready!(self.matrix_client.poll()) {
-                self.handle_sync_response(sync_response);
-            } else {
-                self.closed = true;
-                self.stop();
-                return Ok(Async::Ready(()));
-            }
+    pub async fn poll_matrix(&mut self) -> Result<(), Error> {
+        while let Some(response) = self.matrix_client.as_mut().next().await {
+            let response = response?;
+            self.handle_sync_response(response).await;
         }
+        Ok(())
     }
 }
 
-impl<IS: AsyncRead + AsyncWrite> TaskExecutor for Bridge<IS> {
-    type Error = io::Error;
-
-    fn task_executor_mut(&mut self) -> &mut TaskExecutorQueue<Self, io::Error> {
-        &mut self.executor_queue
+quick_error! {
+    #[derive(Debug)]
+    pub enum Error {
```

Comment:
Some of these are probably unnecessary, ill have to check since this was a copy paste from the matrix module